napari/layers/labels/_labels_key_bindings.py

```diff
@@ -78,11 +78,39 @@ def redo(layer):
 
 @Labels.bind_key('Shift')
 def preserve_labels(layer):
-    """Toggle preserve label option when pressed"""
+    """Toggle preserve label option when pressed."""
     # on key press
     layer.preserve_labels = not layer.preserve_labels
 
     yield
 
     # on key release
     layer.preserve_labels = not layer.preserve_labels
+
+
+@Labels.bind_key('Control')
+def switch_fill(layer):
+    """Switch to fill mode temporarily when pressed."""
+    previous_mode = layer.mode
+
+    # on key press
+    layer.mode = Mode.FILL
```

Comment:
alternatively, I gave it a try using double click to switch to fill, but it is quite some work, because when the first click always triggers a paint, the area would be painted with new label, and the second click tries to fill the area already using the new label, hence not filling the broader area of old label as intended.

Reply:
yeah, let's stay away from double click for now

Reply:
This also doesn't seem to work for me. There maybe some problem binding to a modifier? Unclear, i can look in more. I think @jni may have run into problems here too in another context

Reply:
I was able to use control to swap the mode as expected, wonder if our mac key binding is any different? I have been swapping cmd with ctrl as well